modules/cudf/src/dataframe/concat.ts

```diff
@@ -107,7 +107,13 @@ export function concat<TFirst extends DataFrame, TRest extends DataFrame[]>(firs
 
   type TResultTypeMap = ConcatTypeMap<TFirst, TRest>;
 
-  return new DataFrame(names.reduce(
-    (map, name, index) => ({...map, [name]: Series.new(concatenatedTable.getColumnByIndex(index))}),
-    {} as SeriesMap<{[P in keyof TResultTypeMap]: TResultTypeMap[P]}>));
+  // clang-format off
+  return new DataFrame(
+    names.reduce((map, name, index) =>
+    ({...map, [name]: Series.new(concatenatedTable.getColumnByIndex(index))}),
+    {} as SeriesMap<{[P in keyof TResultTypeMap]: TResultTypeMap[P]}>)
+  ) as TResultTypeMap[keyof TResultTypeMap] extends never
+    ? never
+    : DataFrame<{[P in keyof TResultTypeMap]: TResultTypeMap[P]}>;
+  // clang-format on
```

Comment:
Do we want to store `{[P in keyof TResultTypeMap]: TResultTypeMap[P]}` as a type? then use it on both line 114 and line 117? That's about it in terms of trying to condense this down, not sure if there are any other neat treats we can use here.

Reply:
The type `{[P in keyof TResultTypeMap]: TResultTypeMap[P]}` seems fine to me here as-is.

We throw an error in the case the column types can't be concatenated, right?

Reply:
Discussed this, findcommontype already throws.